src/scripts/basicContext.js

```diff
@@ -1,264 +1,266 @@
-let overflow = null
+window.basicContext = function() {
```

Comment:
Just a quick feedback before I call it a day. This seems to be really old-school: Extending the global `window` object by a function which contains everything. I don't know if this is a requirement for Vibe, but I doubt it (otherwise Vibe is not an option). I would consider this a real step back.

Either we leave it as before or we put everything into a proper JS module.

Reply:
> I know that this is a bad idea, but does this work with Gulp in the frontend?

It does. Just have a look at how basicModal works.

> Does Gulp automatically change the exported structs and makes them usable in "normal" JS?

Not natively, but that is what `gulp-browserify` and/or `gulp-babel` are for. They "fake" module support for browsers that do not support modules. Otherwise we could not even use any of the other proper NPM modules int the frontend, but we do.

That is why we have an array `js` and an array `scripts` in

https://github.com/LycheeOrg/Lychee-front/blob/a89efe2b59da434dcf3a933d5a3318a4c1edae5f/gulpfile.js#L85-L120

The name of the arrays is not helpful, but here is what happens.

 1. The array `js` contains all old-style plain JS files.
 2. The gulp task `main--js` simply concatenates all the files from `js` into a single, temporary output file `_main--javascript.js`.
 3. The array `scripts` contains all JS modules from NPM plus the temporary file `_main--javascript.js` from the previous step
 4. The gulp-task `main--scripts` takes all the necessary NPM modules plus the temporary file and somehow merges it into something which browsers do understand

I have not yet fully understand how it is working and I prefer no to, because Gulp is piece of undocumented shit (sorry, for the harsh words). (Everytime I try to change something, add a file, etc., Gulp totally breaks and it is completely unclear to me why.) However, I know that it somehow takes NodeJS modules and makes them browser compatible. I found that out when I once tried to streamline that mess and got an error in my browser that the keyword `module` is not implemented by browsers.

Otherwise, I would simply suggest to take the build script https://github.com/LycheeOrg/basicModal/blob/master/gulpfile.js as a role model. If you have a look at https://github.com/LycheeOrg/basicModal/blob/master/src/scripts/main.js you will see that we are able to use `export` on some methods and we do not use `export` on other methods. "Somehow" only the exported methods are visible in `basicModal` through some magic (again that has something to do with `babel`  and `browserify`).


`scripts` include all Javascript files which are proper JS modules,

Reply:
> But that's using Gulp I wanted to avoid...

I would like to avoid Gulp, too, because it seems to be abandoned and is (I repeat) a pile of undocumented shit.

I also considered switch to Webpack or just use plain Node scripts. I wrote so the day before yesterday in Gitter. However, after four hours of studying the Webpack documentation I have up on Webpack, because it the documentation has the same crappy quality as Gulp. They always provide you with unhelpful tutorials which only show you the one thousandth variation on how to change the title of your web application (thank you, I know that), but they never manage to _explain_ things.

Anyway, yes Gulp has security flaws. But luckily they only affect the build system, not the productive code. Hence, they are not nice, but somehow tolerable. Yes, Gulp is abandoned and undocumented. Yes, I would like to replace Gulp.

But unless we find a working alternative, which provides the _same features_, we should stay with Gulp. Don't get me wrong, I would be totally fine with Webpack, Vite, Grunt, any other build tool, or plain Node scripts, whatever you like. But it must provide the features we currently use. Otherwise they are not a viable alternative.

I don't have the time to dive into that problem deeply neither do I feel like it. I would rather like to work on the frontend and do things that matter, then spending my time on the build ecosystem. That is why I decided for myself to stay with Gulp for the moment, despite the annoying security warnings, because it simply does what it is supposed to do.

Reply:
> But what should happen with this now?

Frankly, I don't understand your question or more precisely, I don't know what you are up to with this question.

We had (and have) a working build system based on Gulp. Gulp isn't perfect, it is undocumented, seems to be abandoned upstream, etc. pp., but it _works_.

> I agree and for me, docs of Vite weren't good, too. [...] Is there an alternative way of providing the object in a more clean way? My only idea which could change this is using `module.exports`, which might be better for the code quality, but I believe this didn't work in the frontend (I'm not sure though).

Again, I don't understand what you want to tell me. I am totally with you. The documentation of all these build tools (Gulp, Grunt, Webpack, Vite, you name it) seem to be lacking. As I wrote previously, I spent four hours with reading documentation of Webpack before I decided to put it aside (for now). For me, there are currently more important issues than spending time on the build system as long as we have a working solution which simply _does the job_.

Eventually, we will have to replace Gulp. Moreover, I fear the task will fall back on me even though I don't want to.

If you want to do it, be my guest. I won't stop you and I admire your dedication. If I don't have to do deal with a better build system myself, it would give me a feeling of relief. But please don't ask me questions, how to do _X_ or tell me that your are not sure if _Y_ actually works. Otherwise I could do it myself in the first place, which I don't want to do right now. Either you come up with a working alternative solution which provides all the features we had with Gulp and which we need, or you leave it.

> Migrating the frontend is not easy, and I'm also not sure if we should do it.

Nobody said it would be easy. (Otherwise I would not have spent half a unproductive day myself with Webpack.) However, this is your PR and it was your idea to go for Vite. Nobody asked you to use Vite or replace Gulp. I only pointed out that certain things are lacking. So either you figure out how to fix them or you drop it, but don't expect me to step in and tie up the loose ends.